src/github.com/getlantern/balancer/balancer.go

```diff
@@ -49,7 +68,25 @@ func New(dialers ...*Dialer) *Balancer {
 // remaining Dialers until it either manages to connect, or runs out of dialers
 // in which case it returns an error.
 func (b *Balancer) DialQOS(network, addr string, targetQOS int) (net.Conn, error) {
-	dialers := b.dialers
+	var dialers []*dialer
+
+	// Checking destination port.
+	_, port, _ := net.SplitHostPort(addr)
+
+	// Are we attempting to connect to port 80 (plain HTTP)?
```

Comment:
> I.e. only proxying port 80 on our servers wouldn't mitigate the attack since the attack is typically over port 80.

Yes it would, because censors would not have any chance to tamper with plain HTTP resources at all: any HTTP resource would be intercepted and proxied to a trusted proxy over TLS:

Client -> Plaintext HTTP request (port 80) -> Proxy (over TLS) -> Trusted proxy -> facebook:80

If the censors are between Trusted proxy and facebook:80 then yes, they could inject scripts on facebook.com and feed them to our trusted proxy. But if censors are between client and trusted proxy, then they would not be able to see or tamper any HTTP traffic at all.

In particular with the trust flag, we'll take care of cases like this one:

Client -> Plaintext HTTP request (port 80) -> Proxy (over TLS) -> Bad proxy

In this case, Bad proxy could write anything it wants back to the client, which does not have any way to notice if someone tampered with the content.

But, we are still allowing potentially Bad Proxies to exists, they could be handy too:

Client -> HTTPs request (port 443) -> Proxy (over TLS) -> Bad proxy -> Internet -> Facebook:443

This is a completely different case, if Bad Proxy tampers with the content, the SSL connection would not be valid, since for tampering it would be required to decipher, inject evil stuff, and cipher the content again with a rogue certificate, the browser would alert the user of such situation.